pkg/jobconfig/files.go

```diff
@@ -499,3 +474,81 @@ func MakeRegexFilenameLabel(possibleRegex string) string {
 	}
 	return label
 }
+
+func isStale(job prowconfig.JobBase, newLabel ProwgenLabel) bool {
+	genLabel, generated := job.Labels[ProwJobLabelGenerated]
+	return generated && genLabel != string(newLabel)
+}
+
+func isGenerated(job prowconfig.JobBase) bool {
+	_, generated := job.Labels[ProwJobLabelGenerated]
+	return generated
+}
+
+func PruneStaleJobs(jobDir string, oldLabel, newLabel ProwgenLabel) error {
+	if err := OperateOnJobConfigDir(jobDir, func(jobConfig *prowconfig.JobConfig, info *Info) error {
+		if info.Type == "periodics" {
+			return nil
+		}
+
+		pruned := prowconfig.JobConfig{
+			Presubmits:  map[string][]prowconfig.Presubmit{},
+			Postsubmits: map[string][]prowconfig.Postsubmit{},
+		}
+
+		for repo, jobs := range jobConfig.Presubmits {
+			for _, job := range jobs {
+				if isStale(job.JobBase, newLabel) {
+					continue
+				}
+
+				if isGenerated(job.JobBase) {
+					job.Labels[ProwJobLabelGenerated] = string(oldLabel)
+				}
+
+				pruned.Presubmits[repo] = append(pruned.Presubmits[repo], job)
+			}
+		}
+
+		for repo, jobs := range jobConfig.Postsubmits {
+			for _, job := range jobs {
+				if isStale(job.JobBase, newLabel) {
+					continue
+				}
+				if isGenerated(job.JobBase) {
+					job.Labels[ProwJobLabelGenerated] = string(oldLabel)
+				}
+
+				pruned.Postsubmits[repo] = append(pruned.Postsubmits[repo], job)
+			}
+		}
+
+		for repo, jobs := range pruned.Presubmits {
```

Comment:
You only add to `pruned.Presubmits` when you have a reason to so I don't think you'll ever hit a case where you have an empty slice?

Reply:
Right. A remnant from previous solution :)